src/gluonts/dataset/parallelized_loader.py

```diff
@@ -84,16 +85,77 @@ def reduce_ndarray(data):
 ForkingPickler.register(nd.NDArray, reduce_ndarray)
 
 
+def _is_stackable(
+    arrays: List[Union[np.ndarray, mx.nd.NDArray, Any]],
+) -> bool:
+    """
+    Check if elements are scalars, have too few dimensions, or their
+    time axes have equal length; i.e. they are directly `stack` able.
+    """
+    if isinstance(arrays[0], (mx.nd.NDArray, np.ndarray)):
+        s = set(arr.shape[0] for arr in arrays)
+        return len(s) <= 1 and arrays[0].shape[0] != 0
+    return True
+
+
+def _pad_arrays(
+    data: List[Union[np.ndarray, mx.nd.NDArray]], pad_axis: int = 0,
+) -> List[Union[np.ndarray, mx.nd.NDArray]]:
+    assert isinstance(data[0], (np.ndarray, mx.nd.NDArray))
+    is_mx = isinstance(data[0], mx.nd.NDArray)
+
+    # MxNet causes a segfault when persisting 0-length arrays. As such,
+    # we add a dummy pad of length 1 to 0-length dims.
+    max_len = max(1, functools.reduce(max, (x.shape[pad_axis] for x in data)))
+    padded_data = []
+
+    for x in data:
+        # MxNet lacks the functionality to pad n-D arrays consistently.
```

Comment:
So `mxnet.ndarray.pad`: https://beta.mxnet.io/api/ndarray/_autogen/mxnet.ndarray.pad.html is not an option here?

Reply:
From the docs:
```
Current implementation only supports 4D and 5D input arrays with padding applied only on axes 1, 2 and 3. Expects axes 4 and 5 in pad_width to be zero.
```
🤷‍♂️

Reply:
Ok, super weird.

Reply:
Can you please test whether the workaround described here: https://mxnet.apache.org/versions/1.4.1/tutorials/gluon/gotchas_numpy_in_mxnet.html
makes sense?

Reply:
So I implemented this, see latest commit `c10ded2`.

Tests pass locally, but still fail with a segfault (?). If you also can't spot the problem in the code, I suggest we go back to the old solution. This should only be used by TPPs for a long time, and we can try to work around it.

Reply:
I see, then we'll have to look into it some other time. Anyways, thx for trying!